anker/src/state.rs

```diff
@@ -235,6 +272,102 @@ impl Anker {
+        let token_swap = spl_token_swap::state::SwapV1::unpack(&accounts.pool.data.borrow()[1..])?;
+
+        // Check UST token accounts.
+        self.check_ust_reserve_address(anker_program_id, accounts.anker.key, accounts.ust_token)?;
+
+        // `token_a` should be stSOL.
+        if &token_swap.token_a != accounts.st_sol_token.key {
+            msg!(
+            "Token Swap StSol token is different from what is stored in the instance, expected {}, found {}",
+            token_swap.token_a,
+            accounts.st_sol_token.key
+        );
+            return Err(AnkerError::WrongSplTokenSwapParameters.into());
+        }
+        // `token_b` should be UST.
+        if &token_swap.token_b != accounts.ust_token.key {
+            msg!(
+            "Token Swap UST token is different from what is stored in the instance, expected {}, found {}",
+            token_swap.token_b,
+            accounts.ust_token.key
+        );
+            return Err(AnkerError::WrongSplTokenSwapParameters.into());
+        }
+        // Check pool mint.
+        if &token_swap.pool_mint != accounts.pool_mint.key {
+            msg!(
+            "Token Swap mint is different from what is stored in the instance, expected {}, found {}",
+            token_swap.pool_mint,
+            accounts.pool_mint.key
+        );
+            return Err(AnkerError::WrongSplTokenSwapParameters.into());
+        }
+        // Check stSOL mint.
+        if &token_swap.token_a_mint != accounts.st_sol_mint.key {
+            msg!(
+            "Token Swap StSol mint is different from what is stored in the instance, expected {}, found {}",
+            token_swap.token_a_mint,
+            accounts.st_sol_mint.key
+        );
+            return Err(AnkerError::WrongSplTokenSwapParameters.into());
+        }
+        // Check UST mint.
+        if &token_swap.token_b_mint != accounts.ust_mint.key {
+            msg!(
+            "Token Swap UST mint is different from what is stored in the instance, expected {}, found {}",
+            token_swap.token_b_mint,
+            accounts.ust_mint.key
+        );
+            return Err(AnkerError::WrongSplTokenSwapParameters.into());
+        }
+        // Check pool fee.
+        if &token_swap.pool_fee_account != accounts.pool_fee_account.key {
+            msg!(
+            "Token Swap fee account is different from what is stored in the instance, expected {}, found {}",
+            token_swap.pool_fee_account,
+            accounts.pool_fee_account.key
+        );
+            return Err(AnkerError::WrongSplTokenSwapParameters.into());
+        }
+
+        // Check rewards destination.
+        // The reserve address is checked in `deserialize_anker`, this function
+        // should be called prior to this. We don't need to check the reserve
+        // authority, as the transaction will fail if a different one is provided.
+        if &self.rewards_destination != accounts.rewards_destination.key {
```

Comment:
Isn’t the rewards destination the `ust_reserve` now? Let’s call it like that to avoid confusion.

Reply:
Yeah, but the Terra rewards destination should also exist in the state. I'll rename them accordingly.